### ISSUE TITLE
Max display size (or alternative display) for centroid-and-size-only (CASO) annotation elements

### ISSUE THREAD

**Issue by kheffah** [DigitalSlideArchive/HistomicsTK#731]
This is a lower priority issue, but added here for reference when there is a time for it. The CASO elements clutter the underlying slide when they are bigger than a certain size, see below:

![image](https://user-images.githubusercontent.com/22067552/66707748-b8142400-ed13-11e9-91bf-2941a6dde4c1.png)

This makes it hard to review slides with numerous large polygons. I suggest one of two fixes:

1. (Preferred) - If the annotation is polygonal/rectangular/line, show a line along either the major axis of the minimum bounding ellipse, or -more conveniently-, the diagonal of the minimum bounding box. 

2- (Less preferred) - If the CASO size exceeds a certain size threshold, show a maximum size circle and do not exceed it.

I think option 1 is preferred because it reduces clutter and encodes both "directionality" and size, unlike circles which only encode size.

**Comment by manthey** [comment]
We currently store a bounding box; we could show the ellipse that fills this box, *but* this isn't the minimum bounding ellipse.  The axes would always be aligned with the image edges, and there might be points outside of the ellipse.

We could also compute the best-fit ellipse.  I'd probably want to do that on data ingest, as it could be an expensive problem.  

Instead of sending a centroid and size, we'd send a centroid, a major axis, minor axis, and rotation.  This would increase the amount of data we have to send to the client by two values per element.

**Comment by kheffah** [comment]
@manthey I agree, showing a best-fit ellipse is a better option, especially if (as you suggest) it is calculated during data ingest to avoid display delays. The fact that we need to send two extra values per element is negligible, in my opinion because:

1- This issue is only relevant for large polygons. i.e. When viewing an image at **low magnification**, large irregular polygons are currently displayed as very large circles. If instead, they are shown as a best-fit ellipse this would be resolved. A large enough polygon will have thousands of vertices, so replacing it with four data point is a major performance boost. 

2- If a polygon is below a certain size threshold, the difference between a circle, an ellipse and even a point at low magnification will be negligible. So we could just display the centroid or circle if the polygon is small enough since one would need to zoom anyways to see it. 

Here is one sample use case where the circle "distorts" the perceived underlying polygon:

**Low magnification:**
![image](https://user-images.githubusercontent.com/22067552/67125957-ee461d80-f1c3-11e9-94af-9a92407ff7a1.png)

**High magnification:**
![image](https://user-images.githubusercontent.com/22067552/67126017-0ae25580-f1c4-11e9-8977-5d0c1c7b51d2.png)
